src/core/api/index.js

```diff
@@ -162,111 +161,22 @@ export function callApi({
-|};
-
-export function fetchAddon({ api, slug }: FetchAddonParams) {
-  return callApi({
-    endpoint: `addons/addon/${slug}`,
-    schema: addon,
-    auth: true,
-    state: api,
-  });
-}
-
-type LoginParams = {|
-  api: ApiStateType,
-  code: string,
-  state: string,
-|};
-
-export function login({ api, code, state }: LoginParams) {
-  const params = { config: undefined };
-  const configName = config.get('fxaConfig');
-  if (configName) {
-    params.config = configName;
-  }
-  return callApi({
-    endpoint: 'accounts/login',
-    method: 'POST',
-    body: { code, state },
-    params,
-    state: api,
-    credentials: true,
-  });
-}
-
-export function startLoginUrl(
-  { location }: {| location: ReactRouterLocation |},
-) {
-  const configName = config.get('fxaConfig');
-  const params = {
-    config: undefined,
-    to: url.format({ ...location }),
-  };
-  if (configName) {
-    params.config = configName;
-  }
-  const query = makeQueryString(params);
-  return `${API_BASE}/accounts/login/start/${query}`;
-}
-
-type FeaturedParams = {|
-  api: ApiStateType,
-  filters: Object,
-  page: number,
-|};
-
-export function featured({ api, filters, page }: FeaturedParams) {
-  return callApi({
-    endpoint: 'addons/featured',
-    params: {
-      app: api.clientApp,
-      ...convertFiltersToQueryParams(filters),
-      page,
-    },
-    schema: { results: [addon] },
-    state: api,
-  });
-}
-
-export function categories({ api }: {| api: ApiStateType |}) {
-  return callApi({
-    endpoint: 'addons/categories',
-    schema: { results: [category] },
-    state: api,
-  });
-}
-
-export function logOutFromServer({ api }: {| api: ApiStateType |}) {
-  return callApi({
-    auth: true,
-    credentials: true,
-    endpoint: 'accounts/session',
-    method: 'DELETE',
-    state: api,
-  });
-}
-
-type AutocompleteParams = {|
-  api: ApiStateType,
-  filters: {|
-    query: string,
-    addonType?: string,
-  |},
-|};
-
-export function autocomplete({ api, filters }: AutocompleteParams) {
-  return callApi({
-    endpoint: 'addons/autocomplete',
-    params: {
-      app: api.clientApp,
-      ...convertFiltersToQueryParams(filters),
-    },
-    state: api,
-  });
-}
-
-export const search = searchApi;
+// Schemas
+// TODO: Remove this when
+// https://github.com/mozilla/addons-frontend/issues/2917 is fixed.
+export const addon = addonApi.addonSchema;
+export const category = categoriesApi.categorySchema;
+
+// Add-on APIs
+export const fetchAddon = addonApi.fetchAddon;
+// Auth APIs
+export const login = authenticationApi.login;
+export const logOutFromServer = authenticationApi.logOutFromServer;
+export const startLoginUrl = authenticationApi.startLoginUrl;
+// Category APIs
+export const categories = categoriesApi.categories;
+// Featured Add-on APIs
+export const featured = featuredApi.featured;
+// Search and Autocomplete APIs
+export const autocomplete = searchApi.autocomplete;
+export const search = searchApi.search;
```

Comment:
Can we stop doing this maybe? See: https://github.com/mozilla/addons-frontend/pull/3027#discussion_r137825532.

Reply:
Yes I would like that... but it was more work especially with normalizr. I'll see about removing it in another commit though!

Reply:
Can't we just add per API `import` where we rely on `core/api`?

Reply:
We can but there are some bugs with normalizr and stuff in the tests. But I'll give it a go again... 😄

Reply:
Okay, I just spent a few hours trying to remove these imports and going through the code/tests changing calls from `import * as api from 'core/api'` to more specific imports but it's a total mess. Even after some basic cleanup there are like 50 failing tests and the causes aren't obvious. I feel like this branch is a definite improvement and moving to importing the functions from each file will be a lot easier once we ditch normalizr (#2917).

So I'm just gonna propose the PR as it is. I think it's a better jumping off point to fix #2917 but I don't want to go down the rabbit hole any further for now.

Reply:
Oh ok, sorry for the extra work 😔

Reply:
Oh, no worries. It was worth trying but I realised eventually it just wasn't worth pursuing right now 😄